src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -188,6 +183,12 @@ private TypeSymbol DecodeType(TypeSymbol type)
                 case SymbolKind.ArrayType:
                     return DecodeArrayType((ArrayTypeSymbol)type);
 
+                case SymbolKind.ErrorType:
```

Comment:
>case SymbolKind.ErrorType: [](start = 16, length = 26)

If error types cause issues, I do not think it is worth continuing decoding when we encountered one (this is true for all other situations where we trying to match attributes to types), even if ```type.IsTupleType``` is true. It is simply not worth the risk (hard to test all possible badness flavors) and the trouble. Error types often do not reflect the "shape" (genericity, type arguments, etc) of the original type. We won't be able to use referencing symbol successfully anyway due to a use-site error. So why bother trying to apply the attributes? I suggest we find a reliable way to abort the encoding/decoding once any error type symbol is encountered.

Reply:
The behavior proposed by this PR is to give up decoding on error types, except when the error type looks good enough (we checked it's arity and structure with `IsTupleType` or `IsTupleCompatible`). Continuing to decode seems safe precisely because the shape looks good enough here.

That said, I'm okay with an alternative fix (return bad metadata symbol whenever decoding failed, which is more aggressive fix, but possibly safer). Let me know what you think

---
In reply to: [362909843](https://github.com/dotnet/roslyn/pull/40704#discussion_r362909843) [](ancestors = 362909843)

Reply:
> Continuing to decode seems safe precisely because the shape looks good enough here.

I think that there is no benefit to continue in general. There are other situations where error types are not tuples and the same issue will very likely arise. The fix, as it stands right now, doesn't address the root cause of the crash, just one particular symptom.   

---
In reply to: [362927256](https://github.com/dotnet/roslyn/pull/40704#discussion_r362927256) [](ancestors = 362927256,362909843)

Reply:
We cannot be sure that encoding always succeeds, so trying to get it succeed is a loosing battle. Therefore, the recovery should be safe. We have a clear indication that calling ```HasUseSiteError``` is not safe, can cause a cycle through base classes. Therefore, we should replace its usage with something safe.

---
In reply to: [362929920](https://github.com/dotnet/roslyn/pull/40704#discussion_r362929920) [](ancestors = 362929920,362927256,362909843)